### PR TITLE
Overhaul of RidgeRegression2FoldCV

### PR DESCRIPTION
**Pull request by agoscinski**
Summary of this PR is contained in commit messages.

----
:books: Documentation preview :books:: https://scikit-matter--205.org.readthedocs.build/en/205/

### CODE REVIEW

File: examples/regression/Ridge2FoldCVRegularization.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+# Comparing skmatter 2-fold CV with sklearn leave-one-out CV
+# ----------------------------------------------------------
+#
+# In linear regression, the complexity of computing the weight matrix is
+# theoretically bounded by the inversion of the covariance matrix.  This is
+# more costly when conducting regularized regression, wherein we need to
+# optimise the regularization parameter in a cross-validation (CV) scheme,
+# thereby recomputing the inverse for each parameter.  scikit-learn offers an
+# efficient leave-one-out CV (LOO CV) for its ridge regression which avoids
+# these repeated computations [loocv]_. Because we needed an efficient ridge that works
```
Comment (SanggyuChong) [inline examples/regression/Ridge2FoldCVRegularization.py:45]: might be nicer to directly link the slides here somewhere (as well as leaving a reference at the bottom)

File: examples/regression/Ridge2FoldCVRegularization.py [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+plt.axhline(np.finfo(np.float64).eps, label="float64 accuracy", color="black")
+plt.ylim(1e-16, 1)
+plt.xlabel("alphas (regularization parameter)")
+plt.ylabel("MSE")
+
+plt.legend()
+plt.show()
+
+
+# %% We can see thet Leave-one-out CV is estimating the error wrong for low
```
Comment (SanggyuChong) [inline examples/regression/Ridge2FoldCVRegularization.py:289]: top line is commented out, should be put back in

File: examples/regression/Ridge2FoldCVRegularization.py [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+# alpha values. That seems to be a numerical instability of the method. If we
+# would have limit our alphas to 1E-10, then LOO CV would have reach similar
+# accuracies as the other Tikhonov methods.
+
+# %% **Important** to note that this is not an fully encompasing comparison
```
Comment (SanggyuChong) [inline examples/regression/Ridge2FoldCVRegularization.py:294]: Same as above

File: examples/regression/Ridge2FoldCVRegularization.py [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+plt.loglog(singular_values)
+plt.title("Singular values of our feature matrix X")
+plt.axvline(effective_rank, label="effective rank", color="black")
+plt.xlabel("index feature")
+plt.ylabel("singular value")
+plt.legend()
+plt.show()
+
+# %%
+# One can see the jump in the singular value distribution. That will be later
```
Comment (SanggyuChong) [inline examples/regression/Ridge2FoldCVRegularization.py:121]: really not sure what "jump" I'm supposed to look at. is it the drop around feature index 50? And if so, what does that signify?

**Review by SanggyuChong — COMMENTED** [review]
Nicely written, @agoscinski ! I can see what you are trying to highlight as the advantage of the `skmatter.linear_model.RidgeRegression2FoldCV `, but struggled to grasp the points at the first glance. Might I offer you the following two points you may want to consider taking into account:

- LOOCV vs 2-fold CV does not seem to be a fair comparison to me. Mathematical detail of not having to perform SVD multiple times in your 2-fold CV seem convincing enough.
- I'm having a hard time what exactly is supposed to be interpreted from the two singular value plots. These could use further clarifications.

File: examples/regression/Ridge2FoldCVRegularization.py [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+# **Important** to note that this is not an fully encompasing comparison
+# covering sufficient enough the parameter space. We just want to note that in
+# cases with high feature size and low effective rank the ridge solvers in
+# skmatter can be numerical more stable and act on a comparable speed.
+
+# %%
+# Cutoff and Tikhonov regularization
+# ----------------------------------
+# When using a hard threshold as regularization (using parameter ``cutoff``),
+# the singular values below :math:`\lambda` are cutted off, the size of the
```
Comment (SanggyuChong) [inline examples/regression/Ridge2FoldCVRegularization.py:275]: cutted --> cut

**Review by SanggyuChong — APPROVED** [review]
I like this much better than the previous version, in that the example primarily focuses on the benefits of `RidgeRegression2FoldCV` more than anything. Great work, @agoscinski. (Found one typo that needs to be corrected ;) )

**Review by PicoCentauri — APPROVED** [review]

**Pull request merged**